src/evo/deterministicmns.cpp

```diff
@@ -708,11 +708,9 @@ void CDeterministicMNManager::UpdatedBlockTip(const CBlockIndex* pindex)
 
 bool CDeterministicMNManager::BuildNewListFromBlock(const CBlock& block, const CBlockIndex* pindexPrev, BlockValidationState& state, const CCoinsViewCache& view, CDeterministicMNList& mnListRet, bool debugLogs)
 {
-    AssertLockHeld(cs);
-
     int nHeight = pindexPrev->nHeight + 1;
 
-    CDeterministicMNList oldList = GetListForBlockInternal(pindexPrev);
+    CDeterministicMNList oldList = GetListForBlock(pindexPrev);
```

Comment:
Unrelated?

Reply:
Related. Internal doesn't do CS, non -internal does, so, it must be changed